commands: add command to wiping cluster (#173)
rook/kubectl-rook-ceph: docs/destroy-cluster.md

# Destroying a Cluster

> When a cluster is no longer needed and needs to be torn down, Rook has a [Cleanup guide](https://rook.io/docs/rook/latest/Getting-Started/ceph-teardown/) with instructions to tear it down cleanly. While following that document is highly preferred, there is a command that will automate the cleanup.

Reviewer: ```suggestion
When a cluster is no longer needed and needs to be torn down, Rook has a [Cleanup guide](https://rook.io/docs/rook/latest/Getting-Started/ceph-teardown/) with instructions to tear it down cleanly. While following that document is highly preferred, there is a command that will automate the cleanup of Rook after applications are removed.
```

> The `destroy-cluster` command is used for destroy Rook CRs (custom resources) and override any finalizers that may be preventing the cleanup.

Reviewer: ```suggestion
The `destroy-cluster` command destroys a Rook cluster and deletes CRs (custom resources) by force removing any finalizers that may be preventing the cleanup.
```

## !!! Warning !!!
**This command is not reversible**, and it will destroy your Rook cluster completely with guaranteed data loss.
Only use this command if you are 100% sure that the cluster and its data should be destroyed.

## `destroy-cluster` command

To destroy a cluster, run the command:
```bash
$ kubectl rook-ceph -n rook-ceph destroy-cluster
Warning: Are you sure you want to destroy the cluster in namespace rook-ceph? If absolutely certain, enter: yes-really-destroy-cluster
```

Any other response will cause the command to abort.

## Example of destroying a cluster

```bash
$ kubectl rook-ceph -n rook-ceph destroy-cluster
Warning: Are you sure you want to destroy the cluster in namespace rook-ceph? If absolutely certain, enter: yes-really-destroy-cluster
yes-really-destroy-cluster 
Info: proceeding
Info: Getting resources kind cephblockpoolradosnamespaces

> Warning: resource cephblockpoolradosnamespaces was not found on the cluster

Reviewer: I believe you changed it so this is an Info message instead of a Warning, right? 
```suggestion
Info: resource cephblockpoolradosnamespaces was not found on the cluster
```

Info: Getting resources kind cephblockpools
Warning: resource cephblockpools was not found on the cluster
Info: Getting resources kind cephbucketnotifications
Warning: resource cephbucketnotifications was not found on the cluster
Info: Getting resources kind cephbuckettopics
Warning: resource cephbuckettopics was not found on the cluster
Info: Getting resources kind cephclients
Warning: resource cephclients was not found on the cluster
Info: Getting resources kind cephclusters

> Warning: resource cephclusters was not found on the cluster

Reviewer: Could you show test output from a cluster that has some resources installed? It's rather boring output if nothing was found to delete.

Info: Getting resource kind cephfilesystemmirrors
Info: resource cephfilesystemmirrors was not found on the cluster
Info: Getting resource kind cephfilesystems
Info: removing resource cephfilesystems: myfs
Info: resource myfs was deleted
Info: Getting resource kind cephfilesystemsubvolumegroup
Info: the server could not find the requested resource: cephfilesystemsubvolumegroup
Info: Getting resource kind cephnfses
Info: resource cephnfses was not found on the cluster
Info: Getting resource kind cephobjectrealms
Info: resource cephobjectrealms was not found on the cluster
Info: Getting resource kind cephobjectstores
Info: removing resource cephobjectstores: my-store
Info: resource my-store was deleted
Info: Getting resource kind cephobjectstoreusers
Info: resource cephobjectstoreusers was not found on the cluster
Info: Getting resource kind cephobjectzonegroups
Info: resource cephobjectzonegroups was not found on the cluster
Info: Getting resource kind cephobjectzones
Info: resource cephobjectzones was not found on the cluster
Info: Getting resource kind cephrbdmirrors
Info: resource cephrbdmirrors was not found on the cluster
Info: removing deployment rook-ceph-tools

> Info: done

Reviewer: Is this example output updated? I saw at least one more line in my test today
```suggestion
Info: waiting to clean up resources
Info: done
```

Info: Getting resource kind cephfilesystemmirrors
Info: resource cephfilesystemmirrors was not found on the cluster
Info: Getting resource kind cephfilesystems
Info: removing resource cephfilesystems: myfs
Info: resource myfs was deleted
Info: Getting resource kind cephfilesystemsubvolumegroup
Info: the server could not find the requested resource: cephfilesystemsubvolumegroup
Info: Getting resource kind cephnfses
Info: resource cephnfses was not found on the cluster
Info: Getting resource kind cephobjectrealms
Info: resource cephobjectrealms was not found on the cluster
Info: Getting resource kind cephobjectstores
Info: removing resource cephobjectstores: my-store
Info: resource my-store was deleted
Info: Getting resource kind cephobjectstoreusers
Info: resource cephobjectstoreusers was not found on the cluster
Info: Getting resource kind cephobjectzonegroups
Info: resource cephobjectzonegroups was not found on the cluster
Info: Getting resource kind cephobjectzones
Info: resource cephobjectzones was not found on the cluster
Info: Getting resource kind cephrbdmirrors
Info: resource cephrbdmirrors was not found on the cluster
Info: removing deployment rook-ceph-tools

> Info: done

Reviewer: The tool should wait to print `done` until the cluster cleanup jobs have completed on each node. Otherwise, the cleanup is not done yet. In my testing, I have realized the job took an extra minute or two to complete. The user really shouldn't try to create a new cluster until that cleanup job is completed, or else the cluster will not start properly.

Author: hey @travisn I added a timeout of two minutes in order to ensure that all the resources are deleted

> $

Reviewer 2: ```suggestion
```
?

Info: removing resource cephclusters: my-cluster
Info: resource "my-cluster" is not yet deleted, applying patch to remove finalizer...
Info: resource my-cluster was deleted
Info: Getting resources kind cephblockpoolradosnamespaces
Info: resource cephblockpoolradosnamespaces was not found on the cluster
Info: Getting resources kind cephblockpools
Info: removing resource cephblockpools: builtin-mgr
Info: resource "builtin-mgr" is not yet deleted, applying patch to remove finalizer...
Info: cephblockpools.ceph.rook.io "builtin-mgr" not found
Info: Getting resources kind cephbucketnotifications
Info: resource cephbucketnotifications was not found on the cluster
Info: Getting resources kind cephbuckettopics
Info: resource cephbuckettopics was not found on the cluster
Info: Getting resources kind cephclients
Info: resource cephclients was not found on the cluster
Info: Getting resources kind cephcosidrivers
Info: resource cephcosidrivers was not found on the cluster
Info: Getting resources kind cephfilesystemmirrors
Info: resource cephfilesystemmirrors was not found on the cluster
Info: Getting resources kind cephfilesystems
Info: removing resource cephfilesystems: myfs

> Info: cephfilesystems.ceph.rook.io "myfs" not found

Reviewer: Seems like a bug that the myfs filesystem was found to remove (line 48), but then it was not found on line 49.

> # Destroy Cluster

Reviewer: ```suggestion
# Destroying a Cluster
```

> This command is used for destroy CRDS (custom resource definitions) created by rook-ceph

Reviewer: ```suggestion
When a cluster is no longer needed and needs to be torn down, Rook has a [Cleanup guide](https://rook.io/docs/rook/latest/Getting-Started/ceph-teardown/) with instructions to tear it down cleanly. While following that document is highly preferred, there is a command that will automate the cleanup. 

The `destroy-cluster` command is used for destroy Rook CRs (custom resources) and override any finalizers that may be preventing the cleanup.
```

## !!! Warning !!!

> **This command is not reversible**, and it will destroy your rook-ceph cluster completely and your data,

Reviewer: ```suggestion

**This command is not reversible**, and it will destroy your Rook cluster completely with guaranteed data loss.
```

> please only use this command if your sure that your data must be destroyed.

Reviewer: ```suggestion
Only use this command if you are 100% sure that the cluster and its data should be destroyed.
```

> ## How to use:

Reviewer: ```suggestion
## `destroy-cluster` command
```

> Just you need to run the command, and you will be asked for the confirmation

Reviewer: ```suggestion
To destroy a cluster, run the command:
```

```bash
$ kubectl rook-ceph -n rook-ceph destroy-cluster

> Are you sure you want to destroy the cluster in namespace rook-ceph? If absolutely certain, enter: yes-really-destroy-cluster| 

Reviewer: ```suggestion
Warning: Are you sure you want to destroy the cluster in namespace rook-ceph? If absolutely certain, enter: yes-really-destroy-cluster
```

```

> You must write exactly this answer **yes-really-destroy-cluster**

Reviewer: ```suggestion
To continue, enter `yes-really-destroy-cluster`
```

> to confirm that you really want to execute the command and destroy

Reviewer: ```suggestion
to confirm that the cluster really should be deleted. Any other response will cause the command to abort. 
```

your cluster, any other response will be rejected and
the process will be stopped even responding yes

> ```bash

Reviewer: suggestion: Let's remove this example, no need to show error cases in the docs.

$  kubectl rook-ceph -n rook-ceph destroy-cluster
Are Are you sure you want to destroy the cluster in namespace rook-ceph? If absolutely certain, enter: yes-really-destroy-cluster
Error: you need to response "yes-really-destroy-cluster" to confirm the cluster deletion
exit status 1

```

> Once you respond with the right answer the process will start, and it will be deleting the CRDS created by rook-ceph

Reviewer: No need for this sentence
```suggestion
```

> 

Reviewer: Let's add a header
```suggestion
## Example of destroying a cluster
```
